xbmc/pvr/windows/GUIWindowPVRBase.cpp

```diff
@@ -271,6 +300,89 @@ bool CGUIWindowPVRBase::OpenChannelGroupSelectionDialog(void)
   return true;
 }
 
+bool CGUIWindowPVRBase::HasChannelGroupsControl()
+{
+  CGUIControl* control = GetControl(CONTROL_LSTCHANNELGROUPS);
+  return control && control->IsContainer();
+}
+
+bool CGUIWindowPVRBase::IsChannelGroupsControlFocused()
+{
+  CGUIControl* control = GetControl(CONTROL_LSTCHANNELGROUPS);
+  return control && control->HasFocus();
+}
+
+void CGUIWindowPVRBase::InitChannelGroupsControl()
+{
+  if (HasChannelGroupsControl())
+  {
+    const CPVRChannelGroupPtr activeGroup = GetChannelGroup();
+    if (activeGroup)
+    {
+      m_channelGroups = CServiceBroker::GetPVRManager().ChannelGroups()->Get(m_bRadio)->GetMembers(true);
+      CFileItemList channelGroupItems;
+      int iItemIndex = -1;
+      int iIndex = 0;
+      for (const auto& group : m_channelGroups)
+      {
+        CFileItemPtr item(new CFileItem(group->GetPath(), true));
+        item->m_strTitle = group->GroupName();
+        item->SetLabel(group->GroupName());
+        channelGroupItems.Add(item);
+
+        if (iItemIndex == -1 && activeGroup && *activeGroup == *group)
+          iItemIndex = iIndex;
+        else
+          ++iIndex;
+      }
+
+      CGUIMessage msg(GUI_MSG_LABEL_BIND, GetID(), CONTROL_LSTCHANNELGROUPS, iItemIndex, 0, &channelGroupItems);
+      OnMessage(msg);
+    }
+  }
+}
+
+bool CGUIWindowPVRBase::SelectActiveChannelGroup()
+{
+  if (HasChannelGroupsControl())
```

Comment:
This will stay as it is. Personal freedom.

Reply:
Sure it's a matter of taste, but it's more readable in that case and you already used early returns in this method.

Reply:
I will change it the other way round and remove the other early return.

Reply:
Why? this will introduces more deep nesting. Sorry I don't know why not return and that's it.

Reply:
Because I don't want it. There is absolutely no sense in this kind of discussion. Its my personal coding style and freedom of choice. There is no right and wrong. Please, stop this.